mathics/builtin/compilation.py

```diff
@@ -191,6 +214,7 @@ class CompiledFunction(Builtin):
 
     """
 
+    attributes = A_HOLD_ALL | A_PROTECTED | A_N_HOLD_ALL | A_READ_PROTECTED
```

Comment:
These attributes are very important: they prevent the evaluation of the original compiled expression.